spring-grpc-spring-boot-autoconfigure/src/main/java/org/springframework/grpc/autoconfigure/common/codec/GrpcCodecConfiguration.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2024-2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.grpc.autoconfigure.common.codec;
+
+import io.grpc.Codec;
+import io.grpc.Compressor;
+import io.grpc.Decompressor;
+import io.grpc.DecompressorRegistry;
+import io.grpc.CompressorRegistry;
+import org.springframework.beans.factory.ObjectProvider;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+/**
+ * The autoconfiguration that contains all codec related beans for clients/servers.
+ *
+ * @author Andrei Lisa
+ */
+@Configuration(proxyBeanMethods = false)
+@ConditionalOnClass(Codec.class)
+public class GrpcCodecConfiguration {
+
+	@Bean
+	public CompressorRegistry compressorRegistry(ObjectProvider<Compressor> compressors) {
```

Comment:
Do you mind removing the public here on these beans?

Reply:
done.